packages/core/admin/admin/src/content-manager/components/DynamicTable/CellContent/RepeatableComponent/index.js

```diff
@@ -0,0 +1,135 @@
+import { getTrad } from '../../../../utils';
+
+import CellValue from '../CellValue';
+
+const Button = styled.button`
+  svg {
+    > g,
+    path {
+      fill: ${({ theme }) => theme.colors.neutral500};
+    }
+  }
+  &:hover {
+    svg {
+      > g,
+      path {
+        fill: ${({ theme }) => theme.colors.neutral600};
+      }
+    }
+  }
+  &:active {
+    svg {
+      > g,
+      path {
+        fill: ${({ theme }) => theme.colors.neutral400};
+      }
+    }
+  }
+`;
+
+const ActionWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: ${32 / 16}rem;
+  width: ${32 / 16}rem;
+  svg {
+    height: ${4 / 16}rem;
+  }
+`;
+
+function getMainFieldValue(field, name) {
+  return field[name] || field.id;
+}
+
+function getMainFieldType(field, name, defaultType) {
+  if (field[name]) {
+    return defaultType;
+  }
+
+  return 'text';
+}
+
+const RepeatableComponentCell = ({ value, metadatas }) => {
+  const [visible, setVisible] = useState(false);
+  const { formatMessage } = useIntl();
+  const buttonRef = useRef();
+  const {
+    mainField: { type: mainFieldType, name: mainFieldName },
+  } = metadatas;
+  const subItems = [...value.slice(1)];
+
+  const handleTogglePopover = () => setVisible(prev => !prev);
+
+  return (
+    <Flex {...stopPropagation}>
+      <Tooltip
+        label={formatMessage({
+          id: getTrad('DynamicTable.component.repeatable.toggle'),
+          defaultMessage: 'Toggle repeatable values',
+        })}
+      >
+        <Button type="button" onClick={handleTogglePopover} ref={buttonRef}>
+          <Flex>
+            <Typography
+              style={{ maxWidth: '252px', cursor: 'pointer' }}
+              textColor="neutral800"
+              ellipsis
+            >
+              <CellValue
+                type={getMainFieldType(value[0], mainFieldName, mainFieldType)}
+                value={getMainFieldValue(value[0], mainFieldName)}
+              />
+            </Typography>
+            <ActionWrapper>
+              <SortIcon />
+
+              {visible && subItems.length > 0 && (
+                <Popover source={buttonRef} spacing={16} centered>
+                  <FocusTrap onEscape={handleTogglePopover}>
+                    <ul>
+                      {subItems.map(item => (
+                        <Box key={item.id} tabIndex={0} padding={3} as="li">
+                          <Typography>
+                            <CellValue
+                              type={getMainFieldType(item, mainFieldName, mainFieldType)}
+                              value={getMainFieldValue(item, mainFieldName)}
+                            />
+                          </Typography>
+                        </Box>
+                      ))}
+                    </ul>
+                  </FocusTrap>
+                </Popover>
+              )}
+            </ActionWrapper>
+          </Flex>
+        </Button>
```

Comment:
I've kept your original code here and looked into the `SimpleMenu`: to me it seemed, that because the items have hover/ focus effects, they indicate that they are clickable. In this case the different components aren't. What do you think? We could potentially add a `disabled` prop (naming tbd) for each `SimpleMenuItem` though ...

Reply:
Since we are spreading the props, I think you already can pass an `aria-disabled` to the `MenuItem` component 👍🏽

Reply:
Thanks again for the suggestions. I've replaced the custom menu in 3560ceecda6ff931b6692439d4a97f02fea5d640. It makes the whole component much cleaner.

However there is one leftover:

![Screenshot from 2022-02-14 09-29-58](https://user-images.githubusercontent.com/2244375/153827923-5f5ead6e-a2d3-4ff7-8b65-1204b558c07b.png)

Even with `aria-disabled` the `MenuItem` keeps it's hover/ focus states. Is that something I'd need to address in the design-system itself?

Reply:
It should be focusable IMO for the keyboard navigation (+ screen readers announce). If you disable the MenuItem, you'll not be able to scan the MenuItem that's why `aria-diabled` should be enough.

Reply:
@HichamELBSI I agree on focus. The main question is about `hover` since this seems to be misleading to me ...

Reply:
I agree on the `hover` 👍🏽  maybe we could remove the hover when aria-disabled is set to `true` 🤔  WDYT?

Reply:
PR for the design-system: https://github.com/strapi/design-system/pull/525